[routing-manager] new APIs to get state and new CLI commands

abtink opened the pull request:
This commit adds new public OT APIs in `border_routing.h` to get the
current state of `RoutingManager` and to get the current favored
on-link prefix.

This commit also updates CLI `br` commands:
- It moves the CLI `br` implementation into own `cli_br` module.
- It adds `br state` to get the current state
- It changes `br omrprefix`, `br onlinkprefix`, and `br nat64prefix`
  to output both local and favored prefixes when no additional arg
  is provided and allow the use of `local` or `favored` extra arg to
  specify the type.
- It updates the documentation of `br` related commands in the source
  code and in `cli/README_BR.md`.

jwhui commented on include/openthread/instance.h:
```diff
@@ -53,7 +53,7 @@ extern "C" {
  *
  */
-#define OPENTHREAD_API_VERSION (288)
+#define OPENTHREAD_API_VERSION (289)
```
Rebase and update.

jwhui commented on src/core/border_router/routing_manager.hpp:
```diff
@@ -226,10 +259,23 @@ class RoutingManager : public InstanceLocator
+    Error GetOnLinkPrefix(Ip6::Prefix &aPrefix) const;
+
+    /**
+     * This method returns the favored discovered on-link prefix for the adjacent infrastructure link.
```
```suggestion
     * This method returns the favored on-link prefix for the adjacent infrastructure link.
```

jwhui commented on src/cli/README.md:
````diff
@@ -367,19 +367,61 @@ Done
 
 ```bash
-> br omrprefix
+> br info
````
Is it useful to dump all the BR information out in one command? I suspect there is more information we might want to include here. How do we decide what to include or what not to include?

abtink replied:
I wanted to get a snapshot of all prefixes. 
But if we change `br omrprefix` to output both local and favored, it can be used instead.
I am ok removing the `info`.

jwhui commented on src/cli/README.md:
````diff
@@ -367,19 +367,61 @@ Done
+`OPENTHREAD_CONFIG_NAT64_BORDER_ROUTING_ENABLE` is required.
+
+```bash
+> br favorednat64prefix
````
Should we have a common prefix and specify local vs. favored using another arg? For example:
```
br nat64prefix favored
```

abtink replied:
Good idea. 
We can have `br omrprefix` on its own to output both "local" and "favored" and then allow specific ones with the second arg.

abtink commented:
In new push, updated the CLI code 
- Moved the `br` related code into its own module `cli_br`
- Allow `favored`, `local` as extra arg for different prefix sub-commands (e.g., `omrprefix local` or `omrprefix favored`)

Vyrastas commented:
I'm checking the CLI import to openthread.io on this, since there are a lot of changes...

@kylorene FYI

Vyrastas reviewed (commented):
Just one thing to fix, otherwise LGTM

abtink commented on src/cli/cli_br.cpp:
```diff
@@ -0,0 +1,540 @@
+ * Done
+ * @endcode
+ * @par api_copy
+ * #otBorderRoutingSetEnabled
```
Sounds good. Changed it in new push.

jwhui commented on src/cli/README.md:
````diff
@@ -367,27 +367,108 @@ Done
 Done
 ```
 
+### br state
````
Is it worth separating these into a separate `README_BORDER_ROUTER.md` doc?

abtink replied:
Good idea. the `diff` will be larger but I think it is ok (since it's already checked/reviewed).

Added `README_BR.md` (used short form `BR` to match the command name) which now conatins all the `br` sub-commands.
